### PR TITLE
Init

### PR DESCRIPTION
**Pull request by shiroinekotfs**
EOS Client initial PR. This version of EOS is reverse-engineered, exclusively for FPTU Can Tho (FUCT), and can only be used for English tests (ENT503, for example).

The special thing included with this version of EOS Client is, that it contains logging modules (incomplete) and can be run only on multiple platforms (untested), by using Wine.

However, due to the rapid changes in the EOS Client versions, especially the private key inside, it is hard to determine how it serves in real-life conditions.

# Assigned engineer(s)

- @anhvlt-2k6 as reviewer and bug reporter
- @shiroinekotfs as an administrator of this repository

### CODE REVIEW

File: EOSClient/EOSClient/AuthenticateForm.cs [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+			{
+				MessageBox.Show("The account is NOT allowed to take the exam!", "Exam Registering", MessageBoxButtons.OK, MessageBoxIcon.Hand);
+			}
+			else if (eOSData.Status == RegisterStatus.LOGIN_FAILED)
+			{
+				MessageBox.Show("Sorry, unable to verify your information. Check [User Name] and [Password]!", "Login failed");
+			}
+			if (eOSData.Status == RegisterStatus.NEW || eOSData.Status == RegisterStatus.RE_ASSIGN)
+			{
+				//Hide();
```
Comment (anhvlt-2k6) [inline EOSClient/EOSClient/AuthenticateForm.cs:298]: Disabled `Hide()` is not recommended. Mouse logging and Tracking have not yet discovered, it is pretty risky if you still do it.

File: EOSClient/EOSClient/AuthenticateForm.cs [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+	}
+
+	private void AuthenticateForm_Load(object sender, EventArgs e)
+	{
+		string text = "EOS_Server_Info.dat";
+		if (File.Exists(Application.StartupPath + "\\" + text))
+		{
+			try
+			{
+				string key = "04021976";
```
Comment (anhvlt-2k6) [inline EOSClient/EOSClient/AuthenticateForm.cs:333]: Key is accepted, so far. But you will need verify for the key bypassing for further investigation. I know what you are planning, after all.

File: EOSClient/EOSClient/Logging.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        {
+            File.Delete(_log);
+            File.WriteAllText(_log, "Starting EOS Client 23.03.20.24 Logging...\n");
+        } catch 
+        {
+            File.WriteAllText(_log, "Starting EOS Client 23.03.20.24 Logging...\n");
+        }
+    }
+
+    public static void LoggingForURL(string url)
```
Comment (anhvlt-2k6) [inline EOSClient/EOSClient/Logging.cs:25]: Why don't we just use other keys for those static logging? Or combine for all-in-one logging procedure?

**Review by anhvlt-2k6 — COMMENTED** [review]
I just leave comments on flags. Hope you can see those as soon as possible.

**Comment by shiroinekotfs** [comment]
Ignored all!

**Pull request merged**